modules/dnn/src/layers/attention_layer.cpp

```diff
@@ -24,6 +24,93 @@ static void packWeight(size_t num_heads, size_t head_size, size_t input_hidden_s
+                           const float* cos_table,
+                           size_t seq_len,
+                           size_t head_size)
+{
+    CV_Assert(head_size % 2 == 0);
+    size_t half_dim = head_size / 2;
+
+    for (size_t pos = 0; pos < seq_len; ++pos)
+    {
+        float*       out_ptr = data     + pos * head_size;
+        const float* sin_ptr = sin_table + pos * half_dim;
+        const float* cos_ptr = cos_table + pos * half_dim;
+
+#if (CV_SIMD || CV_SIMD_SCALABLE)
+        const size_t w = VTraits<v_float32>::vlanes();  // dynamic lanes for RVV
+        size_t d = 0;
+        for (; d + w <= half_dim; d += w)
+        {
+            // load sin/cos into scalable vectors
+            v_float32 sin_v = vx_load(sin_ptr + d);
+            v_float32 cos_v = vx_load(cos_ptr + d);
+
+            // deinterleave
+            v_float32 x_even, x_odd;
+            v_load_deinterleave(out_ptr + 2*d, x_even, x_odd);
+
+            // complex rotation
+            v_float32 out_even = v_sub(v_mul(cos_v, x_even),
+                                    v_mul(sin_v, x_odd));
+            v_float32 out_odd  = v_add(v_mul(sin_v, x_even),
+                                    v_mul(cos_v, x_odd));
+
+            // store back
+            v_store_interleave(out_ptr + 2*d, out_even, out_odd);
+        }
+        // scalar tail
+        for (; d < half_dim; ++d)
+        {
+            float s  = sin_ptr[d];
+            float c  = cos_ptr[d];
+            float xe = out_ptr[2*d];
+            float xo = out_ptr[2*d+1];
+            out_ptr[2*d]   = xe * c - xo * s;
+            out_ptr[2*d+1] = xo * c + xe * s;
+        }
+#else
+        // fallback: pure scalar
+        for (size_t d = 0; d < half_dim; ++d)
+        {
+            float s  = sin_ptr[d];
+            float c  = cos_ptr[d];
+            float xe = out_ptr[2*d];
+            float xo = out_ptr[2*d+1];
+            out_ptr[2*d]   = xe * c - xo * s;
+            out_ptr[2*d+1] = xo * c + xe * s;
+        }
+#endif
+    }
+}
+
+
+static void precompRotationTables(float *data,
+                                  size_t seq_len,
+                                  size_t head_size) {
+    // RoPE precomputation
+    // RoPE is a positional encoding method used in transformer models.
+    // It uses sine and cosine functions to encode the position of tokens in a sequence
+    // initially introduced for NLP in https://arxiv.org/pdf/2104.09864
+
+    // assume data is of shape [2,seq_ken,head_size]
+
+    float* sin_table = data;
+    float* cos_table = sin_table + seq_len * head_size;
+
+    for (size_t pos = 0; pos < seq_len; ++pos) {
+        for (size_t i = 0; i < head_size; ++i) {
+            float theta = pos * std::pow(10000.f, -2.f * static_cast<float>(i) / static_cast<float>(head_size * 2));
+            sin_table[pos * head_size + i] = std::sin(theta);
+            cos_table[pos * head_size + i] = std::cos(theta);
+        }
+    }
```

Comment:
We have `v_sincos` to optimize it: https://docs.opencv.org/5.x/df/d91/group__core__hal__intrin.html#gaebe923bb2935eae727994a359c8c49c0